src/util/misc/vectors.ts

```diff
@@ -0,0 +1,74 @@
+import { sin } from './sin';
+import { cos } from './cos';
+import { IPoint, Point } from '../../point.class';
+import { TRadian } from '../../typedefs';
+
+/**
+ * Rotates `vector` with `radians`
+ * @static
+ * @memberOf fabric.util
+ * @param {Point} vector The vector to rotate (x and y)
+ * @param {Number} radians The radians of the angle for the rotation
+ * @return {Point} The new rotated point
+ */
+export const rotateVector = (vector: Point, radians: TRadian) => {
+  const sinus = sin(radians),
+        cosinus = cos(radians);
+  return new Point(
+    vector.x * cosinus - vector.y * sinus,
+    vector.x * sinus + vector.y * cosinus,
+  );
+};
+
+/**
+ * Creates a vetor from points represented as a point
+ * @static
+ * @memberOf fabric.util
+ *
+ * @param {Point} from
+ * @param {Point} to
+ * @returns {Point} vector
+ */
+export const createVector = (from: IPoint, to: IPoint): Point => new Point(to).subtract(from);
```

Comment:
i don't get this. if the type was Point , way we changed it to interface? we expect devs to use points or objects with x/y?

Reply:
You are right
I committed it and felt the same
Reverr this 
And safeguard createVector instead